### PR TITLE
13 Write `Res` struct

### PR DESCRIPTION
**Pull request by gregor-tokarev**
1. Add reading from buffer for request
2. Add basic implementation of response struct

### CODE REVIEW

**Review by alytidae — APPROVED** [review]

**Review by gr3yknigh1 — CHANGES_REQUESTED** [review]

File: restless/src/response.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        res
+    }
+
+    pub fn status(&'a mut self, status: usize) -> &mut Res {
+        self.status = status;
+
+        self
+    }
+
+    pub fn set(&'a mut self, header_key: &'a str, header_value: &'a str) -> &'a mut Res {
```
[the remark this answers is not in the document]
Comment (gregor-tokarev) [reply restless/src/response.rs:30]: express api enforce this

File: restless/src/response.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        let mut res = Res {
+            status: 200,
+            headers: HashMap::new(),
+            stream,
+        };
+
+        res
+    }
+
+    pub fn status(&'a mut self, status: usize) -> &mut Res {
```
[the remark this answers is not in the document]
Comment (gregor-tokarev) [reply restless/src/response.rs:24]: express api enforce this

File: restless/src/response.rs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+pub struct Res<'a> {
+    stream: WriteHalf<'a>,
+    status: usize,
+    headers: HashMap<&'a str, &'a str>,
+}
+
+static STATUS_TITLES: Lazy<HashMap<usize, &'static str>> = Lazy::new(|| {
+    HashMap::from([
+        (200, "OK"),
```
Comment (gr3yknigh1) [inline restless/src/response.rs:16]: Duplication at 21 line

File: restless/src/response.rs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        (100, "Continue"),
+        (101, "Switching"),
+        (102, "Processing"),
+        (103, "Early"),
+        (200, "OK"),
+        (201, "Created"),
+        (202, "Accepted"),
+        (203, "Non-Authoritative"),
+        (200, "OK"),
```
Comment (gr3yknigh1) [inline restless/src/response.rs:25]: same duplication

**Review by gr3yknigh1 — CHANGES_REQUESTED** [review]
All good. Just remove duplication

**Pull request merged**